mesonbuild/interpreter/interpreter.py

```diff
@@ -830,7 +830,7 @@ def run_command_impl(self,
             elif isinstance(a, mesonlib.File):
                 expanded_args.append(a.absolute_path(srcdir, builddir))
             elif isinstance(a, ExternalProgram):
-                expanded_args.append(a.get_path())
+                expanded_args += a.command
```

Comment:
I suspect this is causing the windows regressions. Just to confirm my theory (since I don't have a windows machine handy), can you try:
```python
expanded_args.append(a.get_path())
expanded_args.extend(a[1:])
```
and see if that makes everything happy?

Reply:
AFAIK, one of the issue is that `a.get_path()` would currently return the second element from the crossfile.